### PR TITLE
`<chrono>` Estimate width of aligned chronat strings

### LINKED ISSUE #2002: <chrono>: incorrect estimated string width when formatting year_month_day into fullwidth character
**Command-line test case**
```
Active code page: 65001

D:\cpptest>type repro.cpp
#include <format>
#include <iostream>
#include <chrono>
#include <locale>
int main()
{
        using namespace std::chrono;
        std::locale loc{ "chinese.UTF-8" };
        std::locale::global(loc);
        const year_month_day date = 2021y / June / 1d;
        std::cout << std::format("+{:-^60}+\n", "");
        std::cout << std::format("|{:^60}|\n", "English & 中文");
        std::cout << std::format("|{:^60%Y %B %d %A}|\n", date);
        std::cout << std::format("|{:^60}|\n", std::format("{:%Y %B %d %A}", date));
        std::cout << std::format("+{:-^60}+\n", "");
        return 0;
}

D:\cpptest>cl /std:c++20 /utf-8 .\repro.cpp
Microsoft (R) C/C++ Optimizing Compiler Version 19.29.30129.3 for x86
Copyright (C) Microsoft Corporation.  All rights reserved.

repro.cpp
D:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.29.30129\include\chrono(2879): warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc
Microsoft (R) Incremental Linker Version 14.29.30129.3
Copyright (C) Microsoft Corporation.  All rights reserved.

/out:repro.exe
repro.obj

D:\cpptest>repro.exe
+------------------------------------------------------------+
|                       English & 中文                       |
|                  2021 六月 01 星期二                  |
|                    2021 六月 01 星期二                     |
+------------------------------------------------------------+
```

**Expected behavior**
I expect the output to be like:
```
+------------------------------------------------------------+
|                       English & 中文                       |
|                    2021 六月 01 星期二                     |
|                    2021 六月 01 星期二                     |
+------------------------------------------------------------+
```
i.e. the two lines of "2021 六月 01 星期二" should have the same width.
**STL version**
```
Microsoft Visual Studio Enterprise 2019 Preview
版本 16.11.0 Preview 1.0

### PR DESCRIPTION
**Pull request by MattStephanson**
Fixes #2002.

### CODE REVIEW

File: tests/std/tests/P0355R7_calendars_and_time_zones_formatting/test.cpp [hunk 1]
```diff
@@ -955,6 +969,14 @@ void test() {
 
     test_zoned_time_formatter<char>();
     test_zoned_time_formatter<wchar_t>();
+
+#if !defined(_DLL) || _ITERATOR_DEBUG_LEVEL == DEFAULT_IDL_SETTING
+    test_locale<wchar_t>();
+#ifndef MSVC_INTERNAL_TESTING // TRANSITION, the Windows version on Contest VMs doesn't always understand ".UTF-8"
+    setlocale(LC_ALL, ".UTF-8");
```
Comment (StephanTLavavej) [inline tests/std/tests/P0355R7_calendars_and_time_zones_formatting/test.cpp:976]: Should technically include `<clocale>` for `setlocale()`.

Comment (StephanTLavavej) [inline tests/std/tests/P0355R7_calendars_and_time_zones_formatting/test.cpp:976]: I think we should also verify that `setlocale()` succeeds:

https://github.com/microsoft/STL/blob/1866b848f0175c3361a916680a4318e7f0cc5482/tests/std/tests/P0645R10_text_formatting_utf8/test.cpp#L157

**Review by StephanTLavavej — COMMENTED** [review]
Looks great! I'll validate and push trivial test changes.

**Review by StephanTLavavej — APPROVED** [review]

**Review by CaseyCarter — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
Thanks for investigating and fixing this bug! :beetle: :heavy_check_mark: :smiley_cat:

**Pull request merged**